app/src/main/java/com/humara/nagar/ui/report/ReportFragment.kt

```diff
@@ -23,23 +59,281 @@ class ReportFragment : BaseFragment() {
+        } catch (e: ActivityNotFoundException) {
+            // display error state to the user
+            Logger.debugLog("Exception caught: $e")
+        }
+    }
+
+    @Throws(IOException::class)
+    private fun createImageFile(): File {
+        // Create an image file name
+        val timeStamp: String = SimpleDateFormat("yyyyMMdd_HHmmss").format(Date())
+        val storageDir: File? =
+            requireActivity().getExternalFilesDir(Environment.DIRECTORY_PICTURES)
+        return File.createTempFile("JPEG_${timeStamp}_", ".jpg", storageDir).apply {
+            // Save a file: path for use with ACTION_VIEW intents
+            currentPhotoPath = absolutePath
+        }
+    }
+
+    private fun getAddress() {
+        try {
+            val addresses = Utils.getAddressFromLongAndLat(requireContext())
+            val address = addresses[0]
+            val addressLine = address.getAddressLine(0)
+            val city = address.locality
+            val state = address.adminArea
+            val country = address.countryName
+            val postalCode = address.postalCode
+            Logger.debugLog("Address:\nAddressLine-> $addressLine\nCity-> $city\nState-> $state\nCountry-> $country\nPostalCode-> $postalCode")
+            binding.inputLocation.setInput(addressLine)
+            reportViewModel.setLocation(addressLine)
+        } catch (e: java.lang.Exception) {
+            Logger.debugLog("Exception caught at getAddress: $e")
+        }
+    }
+
+    private fun checkForLocationPermission() {
+        val permissions = PermissionsUtils.locationPermissions
+        if (!PermissionsUtils.checkPermissions(requireContext(), permissions)) {
+            PermissionsUtils.requestPermissions(this@ReportFragment, permissions)
+            requestedPermissions.clear()
+            requestedPermissions = permissions.toMutableList()
+        } else getAddress()
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+
+        if (requestCode == PICK_IMAGE_MULTIPLE && resultCode == RESULT_OK) {
+            if (data?.clipData != null) {
+                val count = data.clipData!!.itemCount
+                if (count > maxSelection) {
+                    Toast.makeText(
+                        requireContext(),
+                        resources.getString(R.string.imagePickingLimit),
+                        Toast.LENGTH_SHORT
+                    ).show()
+                    return
+                }
+
+                for (i in 0 until count) {
+                    val imageUri = data.clipData!!.getItemAt(i).uri
+                    imageUri?.let { uri ->
+                        imageList.add(uri)
+                        uri.path?.let { it ->
+                            Logger.debugLog("File name: ${File(it).name}")
+                        }
+                    }
+                    imagePreviewAdapter.notifyDataSetChanged()
```

Comment:
we can avoid expensive imagePreviewAdapter.notifyDataSetChanged call here. Discuss over call later

Reply:
Okay.